### PR TITLE
Add additional options to acoustic_model_config builder

### PR DESCRIPTION
**Pull request by SimBe195**
Small extension to `acoustic_model_config` to allow directly setting `state_tying_file`, `phonology.history_length` and `phonology.future_length`.

### CODE REVIEW

**Comment by JackTemaki** [comment]
When you are touching this, can you add types and a dummy docstring? You do not have to fill everything, just what you know.

**Comment by SimBe195** [comment]
> When you are touching this, can you add types (best via typing, not in docstring) and a dummy docstring? You do not have to fill everything, just what you know.

Done.

File: am/config.py [hunk 1]
```diff
@@ -1,24 +1,63 @@
...
+    state_repetitions: int = 1,
+    across_word_model: bool = True,
+    early_recombination: bool = False,
+    tdp_scale: float = 1.0,
+    tdp_transition: Union[TdpValues, Tuple[TdpType, ...]] = (3.0, 0.0, 3.0, 2.0),
+    tdp_silence: Union[TdpValues, Tuple[TdpType, ...]] = (0.0, 3.0, "infinity", 6.0),
+    tying_type: Literal["global", "global-and-nonword"] = "global",
+    nonword_phones: Union[str, List[str]] = "",
+    tdp_nonword: Union[TdpValues, Tuple[TdpType, ...]] = (0.0, 3.0, "infinity", 6.0),
+    state_tying_file: Union[str, tk.Path] = "",
```
Comment (michelwi) [inline am/config.py:32]: can we make this
```suggestion
    state_tying_file: Optional[tk.Path] = None,
```
disallowing `str`. 

And below check for `state_tying_file is not None`

**Review by michelwi — APPROVED** [review]
thanks 👍

**Comment by Marvin84** [comment]
@JackTemaki and @SimBe195 When I merged this change I got the notification that the AppTek hashes are breaking with this change. Isn't this motive of concern?

**Pull request merged**